test/e2e/framework/addon/chart/addon.go

```diff
@@ -0,0 +1,235 @@
+	// `helm install --version {{ChartVersion}}`
+	ChartVersion string
+
+	// Vars are additional --set arguments for helm install
+	// `helm install --set {{Vars[0].Key}}={{Vars[0].Value}} --set {{Vars[1].Key}}={{Vars[1].Value}} ...`
+	Vars []StringTuple
+
+	// Values is a list of paths to additional values.yaml files to include
+	// `helm install --values {{Values[0]}} --values {{Values[1]}} ...`
+	Values []string
+
+	// If UpdateDeps is true, 'helm dep update' will be run against the chart
+	// before installing.
+	// This should only be set to true when the ChartName is a local path on disk.
+	UpdateDeps bool
+}
+
+// StringTuple is a tuple of strings, used to create ordered maps
+type StringTuple struct {
+	Key   string
+	Value string
+}
+
+// Details return the details about the Tiller instance deployed
+type Details struct {
+	// Helm chart release name
+	ReleaseName string
+
+	// Namespace that Tiller has been deployed into
+	Namespace string
+}
+
+func (c *Chart) Setup(cfg *config.Config) error {
+	var err error
+
+	if c.Tiller == nil {
+		return fmt.Errorf("tiller base addon must be provided")
+	}
+	c.tillerDetails, err = c.Tiller.Details()
+	if err != nil {
+		return err
+	}
+
+	c.home, err = ioutil.TempDir("", "helm-chart-install")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Provision an instance of tiller-deploy
+func (c *Chart) Provision() error {
+	err := c.runHelmClientInit()
+	if err != nil {
+		return fmt.Errorf("error running 'helm init': %v", err)
+	}
+
+	if c.UpdateDeps {
+		err := c.runDepUpdate()
+		if err != nil {
+			return fmt.Errorf("error updating helm chart dependencies: %v", err)
+		}
+	}
+
+	err = c.runInstall()
+	if err != nil {
+		return fmt.Errorf("error install helm chart: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Chart) runHelmClientInit() error {
+	err := c.buildHelmCmd("init", "--client-only").Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Chart) runDepUpdate() error {
+	err := c.buildHelmCmd("dep", "update", c.ChartName).Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Chart) runInstall() error {
+	args := []string{"install", c.ChartName,
+		// "--wait",
+		"--namespace", c.Namespace,
+		"--name", c.ReleaseName}
+
+	for _, v := range c.Values {
+		args = append(args, "--values", v)
+	}
+
+	for _, s := range c.Vars {
+		args = append(args, "--set", fmt.Sprintf("%s=%s", s.Key, s.Value))
+	}
+
+	cmd := c.buildHelmCmd(args...)
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Chart) buildHelmCmd(args ...string) *exec.Cmd {
+	args = append([]string{
+		"--home", c.home,
+		"--kubeconfig", c.tillerDetails.KubeConfig,
+		"--kube-context", c.tillerDetails.KubeContext,
+		"--tiller-namespace", c.tillerDetails.Namespace,
+	}, args...)
+	cmd := exec.Command("helm", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd
+}
+
+func (c *Chart) getHelmVersion() (string, error) {
```

Comment:
This function isn't actually used. I'll update it after this merges to actually use it.

Intention is to compare server and client version as part of Provision so we can fail with decent error messages.